libcore/src/routing/ff_router/ffRouter.cpp

```diff
@@ -120,56 +120,86 @@ void FFRouter::CalculateFloorFields()
-    _CroTrByUID.clear();
+    _exitsByUID.clear();
+    _doorByUID.clear();
 
     //get all door UIDs
     std::vector<std::pair<int, int>> roomAndCroTrVector;
 
     for(const auto & [_, trans] : _building->GetAllTransitions()) {
         _allDoorUIDs.emplace_back(trans->GetUniqueID());
-        _CroTrByUID.insert(std::make_pair(trans->GetUniqueID(), trans));
+        _doorByUID.insert(std::make_pair(trans->GetUniqueID(), trans));
         if(trans->IsExit()) {
-            _ExitsByUID.insert(std::make_pair(trans->GetUniqueID(), trans));
+            _exitsByUID.insert(std::make_pair(trans->GetUniqueID(), trans));
         }
         Room * room1 = trans->GetRoom1();
-        if(room1) {
+        if(room1 != nullptr) {
             roomAndCroTrVector.emplace_back(std::make_pair(room1->GetID(), trans->GetUniqueID()));
         }
         Room * room2 = trans->GetRoom2();
-        if(room2) {
+        if(room2 != nullptr) {
             roomAndCroTrVector.emplace_back(std::make_pair(room2->GetID(), trans->GetUniqueID()));
         }
     }
 
     for(const auto & [_, cross] : _building->GetAllCrossings()) {
         _allDoorUIDs.emplace_back(cross->GetUniqueID());
-        _CroTrByUID.insert(std::make_pair(cross->GetUniqueID(), cross));
+        _doorByUID.insert(std::make_pair(cross->GetUniqueID(), cross));
         Room * room1 = cross->GetRoom1();
-        if(room1)
+        if(room1 != nullptr) {
             roomAndCroTrVector.emplace_back(std::make_pair(room1->GetID(), cross->GetUniqueID()));
+        }
     }
 
     if(_hasSpecificGoals) {
         for(auto const & [goalID, goal] : _building->GetAllGoals()) {
-            // TODO add handling for waiting areas
             // TODO add handling for doors with (almost) same distance
             //  ========      =========      =========
             //
             //       ------------------------------
             //       |           goal             |
             //       ------------------------------
-            double minDist = std::numeric_limits<double>::max();
-            int minID      = -1;
-
-            for(auto const & [exitID, exit] : _ExitsByUID) {
-                double dist = goal->GetDistance(exit->GetCentre());
-                if(dist < minDist) {
-                    minDist = dist;
-                    minID   = exitID;
+            // if waiting area, add all transition and crossings (if needed)
+            if(auto * waitingArea = dynamic_cast<WaitingArea *>(goal); waitingArea != nullptr) {
+                auto * room = _building->GetRoom(waitingArea->GetRoomID());
+                std::set<int> doorUIDs;
+
+                if(!_targetWithinSubroom) {
+                    //candidates of current room (ID) (provided by Room)
+                    for(auto transUID : room->GetAllTransitionsIDs()) {
+                        doorUIDs.emplace(transUID);
+                    }
+                    for(auto const & [_, subroom] : room->GetAllSubRooms()) {
```

Comment:
I think we should decide for east const or west const and use it consistently.

Reply:
Changed, to be consistent